### PR TITLE
java: Improve descriptions of `javadoc`

### PR DESCRIPTION
**Pull request by t-kurita**
- Improve descriptions
- Fix some typos
- Remove MPI-1 functions and replace them with MPI-2 functions

Signed-off-by: Kurita, Takehiro <fj6370fp@aa.jp.fujitsu.com>

### CODE REVIEW

**Comment by kawashima-fj** [comment]
ok to test

File: ompi/mpi/java/java/Comm.java [hunk 1]
```diff
@@ -1170,7 +1170,7 @@ public final void abort(int errorcode) throws MPIException
 
 	/**
 	 * Associates a new error handler with communicator at the calling process.
-	 * <p>Java binding of the MPI operation {@code MPI_ERRHANDLER_SET}.
+	 * <p>Java binding of the MPI operation {@code MPI_COMM_ERRHANDLER_SET}.
```
Comment (kawashima-fj) [inline ompi/mpi/java/java/Comm.java]: This should be `MPI_COMM_SET_ERRHANDLER`.

File: ompi/mpi/java/java/Comm.java [hunk 2]
```diff
@@ -1185,7 +1185,7 @@ private native void setErrhandler(long comm, long errhandler)
 
 	/**
 	 * Returns the error handler currently associated with the communicator.
-	 * <p>Java binding of the MPI operation {@code MPI_ERRHANDLER_GET}.
+	 * <p>Java binding of the MPI operation {@code MPI_COMM_ERRHANDLER_GET}.
```
Comment (kawashima-fj) [inline ompi/mpi/java/java/Comm.java]: This should be ` MPI_COMM_GET_ERRHANDLER`.

File: ompi/mpi/java/java/Comm.java (hunk 1 above)
Comment (t-kurita) [reply ompi/mpi/java/java/Comm.java]: I fixed it.

File: ompi/mpi/java/java/Comm.java (hunk 2 above)
Comment (t-kurita) [reply ompi/mpi/java/java/Comm.java]: I fixed it.

**Pull request merged**